File: frontend/src/services/notificationService.js
import api from './api';

export const notificationService = {
  getMyNotifications: async (page = 0, size = 15) => {
    const res = await api.get(`/notifications?page=${page}&size=${size}`);
    return res.data;
  },

  getUnreadNotifications: async () => {
    const res = await api.get('/notifications/unread');
    return res.data;
  },

  getUnreadCount: async () => {
    const res = await api.get('/notifications/unread-count');
    return res.data;
  },

  markAsRead: async (notificationId) => {
    const res = await api.put(`/notifications/${notificationId}/read`);
    return res.data;
  },

  markAllAsRead: async () => {
    const res = await api.put('/notifications/read-all');
    return res.data;
  }
};

File: backend/src/main/java/com/legaltrack/service/NotificationService.java
package com.legaltrack.service;

import com.legaltrack.dto.common.PagedResponse;
import com.legaltrack.dto.notification.NotificationDto;
import com.legaltrack.entity.User;
import com.legaltrack.enums.NotificationType;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface NotificationService {
    void createNotification(User user, NotificationType type, String title, String message, String refType, Long refId, String actionUrl);
    PagedResponse<NotificationDto> getUserNotifications(Long userId, Pageable pageable);
    List<NotificationDto> getUnreadNotifications(Long userId);
    void markAsRead(Long notificationId, Long userId);
    void markAllAsRead(Long userId);
    Long getUnreadCount(Long userId);
}
